src/to-be-visible.js

```diff
@@ -14,10 +14,16 @@ function isStyleVisible(element) {
   )
 }
 
+function hasElementAVisibleAttribute(element) {
```

Comment:
I'd recommend renaming this to `isDetailsElementOpen`.

Reply:
It's checking if the element is invisible due to certain HTML attributes such as `hidden` or `open`. It's actually incorrect since this would consider `<details hidden open>` as being visible.

I reconsidered landing this as-is. I think checking for `!(element.hidden || element.nodeName === 'DETAILS' && !element.open)` is the correct way to handle every combination

Reply:
I still prefer the more descriptive "isDetailsElementOpen". It describes what the function does, not how it does it.

Reply:
It also reflects the fact that this operates specifically on a `<details />` element.

Reply:
Hi, I think it has the responsibility to operates in any element to check for an attribute about the element visibility, just like @eps1lon has mentioned. I've to check the `details` first because of it operates on an exception about its visibility attribute (it doesn't use the `hidden`). But, it's up to you...

> I think checking for `!(element.hidden || element.nodeName === 'DETAILS' && !element.open)` is the correct way to handle every combination

I'll make a spec to ensure that.

Reply:
Got it. Now I understand where my reasoning was flawed. Thanks to you two for clearing that up.

Now, I'll slightly insist in calling the function by what it does. I do not care if it checks for attributes. In my mind this function checks if the given element is considered visible. Isn't it?

Reply:
I've renamed to a name that matches the pattern applied to check the style visibility. Now we have something like that:
```js
isElementVisible => isStyleVisible && isAttributeVisible
```